packages/terra-tabs/src/_CollapsibleTabs.jsx

```diff
@@ -73,6 +73,13 @@ class CollapsibleTabs extends React.Component {
     this.handleSelectionAnimation();
   }
 
+  componentWillReceiveProps(nextProps) {
```

Comment:
Let's switch this over to use `getDerivedStateFromProps` instead of `componentWillReceiveProps`. https://hackernoon.com/replacing-componentwillreceiveprops-with-getderivedstatefromprops-c3956f7ce607

You'll need to use https://github.com/reactjs/react-lifecycles-compat to polyfill `getDerivedStateFromProps` as well. Set that packages as a dependency and import `polyfill` and wrap the `CollapsibleTabs` component with it.

Reply:
I had tried to avoid `componentWillReceiveProps`. The values stored in the internal state were moved out of it in this fix https://github.com/cerner/terra-core/pull/1658. Since `getDerivedStateFromProps` cannot access `this`, these values will have to be moved back inside the local state for `getDerivedStateFromProps` to work.

Reply:
Adding componentWillReceiveProps adds tech debt for us to resolve when we update to React 17 as `componentWillReceiveProps` is being deprecated in react. Instead of trying to make this component act as both and uncontrolled and controlled component via `componentWillReceiveProps/getDerivedStateFromProps` I'd suggest we go back and work on a tech design on https://github.com/cerner/terra-core/issues/1627 for making a controlled version of tabs which could support adding additional tabs.